sdk/tests/SerializationAndDeserializationTests.cc

```diff
@@ -0,0 +1,199 @@
+/*-
+ *
+ * Hedera C++ SDK
+ *
+ * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "AccountId.h"
+#include "ContractId.h"
+#include "ExchangeRate.h"
+#include "ExchangeRateSet.h"
+#include "Hbar.h"
+#include "TransactionId.h"
+#include "TransactionReceipt.h"
+#include "Transfer.h"
+
+#include "impl/TimestampConverter.h"
+
+#include <chrono>
+#include <gtest/gtest.h>
+#include <proto/basic_types.pb.h>
+#include <proto/exchange_rate.pb.h>
+
+using namespace Hedera;
+
+class SerializationAndDeserializationTests : public ::testing::Test
+{
+protected:
+  [[nodiscard]] inline const uint64_t getTestShardNum() const { return mShardNum; }
+  [[nodiscard]] inline const uint64_t getTestRealmNum() const { return mRealmNum; }
+  [[nodiscard]] inline const uint64_t getTestContractNum() const { return mContractNum; }
+  [[nodiscard]] inline const AccountId& getTestAccountId() const { return mAccountId; }
+
+private:
+  const uint64_t mShardNum = 1;
+  const uint64_t mRealmNum = 2;
+  const uint64_t mContractNum = 3;
+  const AccountId mAccountId = AccountId(10ULL);
+  const std::chrono::system_clock::time_point mValidStart = std::chrono::system_clock::time_point();
+};
+
+TEST_F(SerializationAndDeserializationTests, SerializeAccountIdToProtobufTest)
+{
+  // Given
+  const std::string testAccountIdStr = "111.222.333";
+  AccountId testAccountId(testAccountIdStr);
+
+  // When
+  auto protoAccountId = std::unique_ptr<proto::AccountID>(testAccountId.toProtobuf());
+
+  // Then
+  EXPECT_EQ(protoAccountId->shardnum(), testAccountId.getShardNum());
+  EXPECT_EQ(protoAccountId->realmnum(), testAccountId.getRealmNum());
+  EXPECT_EQ(protoAccountId->accountnum(), testAccountId.getAccountNum());
+}
+
+TEST_F(SerializationAndDeserializationTests, DeserializeAccountIdFromProtobufTest)
```

Comment:
To write this test I used test [ProtobufAccountId](https://github.com/hashgraph/hedera-sdk-cpp/blob/c54139e8bc8f507e1f08d052b15eec3d188a0bd0/sdk/tests/AccountIdTest.cc#L125) from `AccountIdTest`.

Reply:
Test `SerializeAccountIdToProtobufTest` & `DeserializeAccountIdFromProtobufTest` moved into `AccountIdTest.cc` in commit [11ed0182be6d48d340a9897994290084e66cf20e].(https://github.com/hashgraph/hedera-sdk-cpp/pull/176/commits/11ed0182be6d48d340a9897994290084e66cf20e)